lib/streamlit/config.py

```diff
@@ -1201,7 +1202,31 @@ def on_config_parsed(func, force_connect=False):
         func()
 
 
+def _validate_theme() -> None:
+    optional_theme_options = {"name", "setAsDefault", "font"}
+    reserved_theme_names = {"auto", "dark", "light"}
+
+    theme_opts = get_options_for_section("customTheme")
+
+    theme_name = cast(str, theme_opts["name"])
+    if theme_name and theme_name.lower() in reserved_theme_names:
+        raise RuntimeError('customTheme.name cannot be "Auto", "Dark", or "Light".')
+
+    required_opts = set(theme_opts.keys()) - optional_theme_options
+    theme_fully_defined = all([bool(theme_opts[k]) for k in required_opts])
+    no_theme_defined = all([not bool(theme_opts[k]) for k in required_opts])
+
+    if theme_fully_defined or no_theme_defined:
+        return
+    else:
+        raise RuntimeError(
+            "customTheme options only partially defined. To specify a theme, "
+            " please set all required options."
```

Comment:
Product will likely want to review the wording

Reply:
:+1: 

I'll put an item in the backlog to make sure that a wordsmithing review happens before release